extensions/src/test/java/org/pentaho/platform/web/http/api/resources/UserRoleListResourceTest.java

```diff
@@ -126,14 +126,25 @@ public void testGetRoles() throws Exception {
   }
 
   @Test
-  public void testGetAllRoles() throws Exception {
+  public void testGetAllRoles() {
     RoleListWrapper mockWrapper = mock( RoleListWrapper.class );
-    doReturn( mockWrapper ).when( userRoleListServiceMock ).getAllRoles();
+    doReturn( mockWrapper ).when( userRoleListServiceMock ).getAllRoles( false );
 
-    RoleListWrapper testWrapper = userRoleListResource.getAllRoles();
+    RoleListWrapper testWrapper = userRoleListResource.getAllRoles( false );
     assertEquals( mockWrapper, testWrapper );
 
-    verify( userRoleListServiceMock, times( 1 ) ).getAllRoles();
+    verify( userRoleListServiceMock, times( 1 ) ).getAllRoles( false );
+  }
+
+  @Test
+  public void testGetAllRolesWithoutAnonymous() {
```

Comment:
There's also a `UserRoleListServiceTest` class where we should add tests for the new functionality.

Reply:
Addressed